lettermgmt/migrations/8.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# © 2016 Iván Todorovich <ivan.todorovich@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+def migrate(cr, version):
+    if version is None:
+        return
+
+    # Migrate 'type' to 'type_id'
+    cr.execute("""ALTER TABLE res_letter
+                  RENAME COLUMN type to type_id""")
```

Comment:
you should borrow openupgrade's code for renaming a model here: https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L328 to get links from other models (mostly) right after you renamed `letter.class`